Validation/EcalHits/src/EcalBarrelSimHitsValidation.cc

```diff
@@ -25,127 +25,84 @@ EcalBarrelSimHitsValidation::EcalBarrelSimHitsValidation(const edm::ParameterSet
+  histo = "EB hits energy spectrum";
+  meEBhitEnergy_ = ib.book1D(histo, histo, 4000, 0., 400.);
 
-    sprintf(histo, "EB hits energy spectrum 2");
-    meEBhitEnergy2_ = dbe_->book1D(histo, histo, 1000, 0., 0.001);
+  histo = "EB hits log10energy spectrum";
+  meEBhitLog10Energy_ = ib.book1D(histo, histo, 140, -10., 4.);
 
-    sprintf(histo, "EB crystal energy spectrum");
-    meEBcrystalEnergy_ = dbe_->book1D(histo, histo, 5000, 0., 50.);
+  histo = "EB hits log10energy spectrum vs normalized energy";
+  meEBhitLog10EnergyNorm_ = ib.bookProfile(histo, histo, 140, -10., 4., 100, 0., 1.);
 
-    sprintf(histo, "EB crystal energy spectrum 2");
-    meEBcrystalEnergy2_ = dbe_->book1D(histo, histo, 1000, 0., 0.001);
+  histo = "EB hits log10energy spectrum vs normalized energy25";
+  meEBhitLog10Energy25Norm_ = ib.bookProfile(histo, histo, 140, -10., 4., 100, 0., 1.);
 
-    sprintf(histo, "EB E1");
-    meEBe1_ = dbe_->book1D(histo, histo, 400, 0., 400.);
+  histo = "EB hits energy spectrum 2";
+  meEBhitEnergy2_ = ib.book1D(histo, histo, 1000, 0., 0.001);
 
-    sprintf(histo, "EB E4");
-    meEBe4_ = dbe_->book1D(histo, histo, 400, 0., 400.);
+  histo = "EB crystal energy spectrum";
+  meEBcrystalEnergy_ = ib.book1D(histo, histo, 5000, 0., 50.);
 
-    sprintf(histo, "EB E9");
-    meEBe9_ = dbe_->book1D(histo, histo, 400, 0., 400.);
+  histo = "EB crystal energy spectrum 2";
+  meEBcrystalEnergy2_ = ib.book1D(histo, histo, 1000, 0., 0.001);
 
-    sprintf(histo, "EB E16");
-    meEBe16_ = dbe_->book1D(histo, histo, 400, 0., 400.);
+  histo = "EB E1";
+  meEBe1_ = ib.book1D(histo, histo, 400, 0., 400.);
 
-    sprintf(histo, "EB E25");
-    meEBe25_ = dbe_->book1D(histo, histo, 400, 0., 400.);
+  histo = "EB E4";
+  meEBe4_ = ib.book1D(histo, histo, 400, 0., 400.);
 
-    sprintf(histo, "EB E1oE4");
-    meEBe1oe4_ = dbe_->book1D(histo, histo, 100, 0.4, 1.1);
+  histo = "EB E9";
+  meEBe9_ = ib.book1D(histo, histo, 400, 0., 400.);
 
-    sprintf(histo, "EB E1oE9");
-    meEBe1oe9_ = dbe_->book1D(histo, histo, 100, 0.4, 1.1);
+  histo = "EB E16";
+  meEBe16_ = ib.book1D(histo, histo, 400, 0., 400.);
 
-    sprintf(histo, "EB E4oE9");
-    meEBe4oe9_ = dbe_->book1D(histo, histo, 100, 0.4, 1.1);
+  histo = "EB E25";
+  meEBe25_ = ib.book1D(histo, histo, 400, 0., 400.);
 
-    sprintf(histo, "EB E9oE16");
-    meEBe9oe16_ = dbe_->book1D(histo, histo, 100, 0.4, 1.1);
+  histo = "EB E1oE4";
+  meEBe1oe4_ = ib.book1D(histo, histo, 100, 0.4, 1.1);
 
-    sprintf(histo, "EB E1oE25");
-    meEBe1oe25_ = dbe_->book1D(histo, histo, 100, 0.4, 1.1);
+  histo = "EB E1oE9";
+  meEBe1oe9_ = ib.book1D(histo, histo, 100, 0.4, 1.1);
 
-    sprintf(histo, "EB E9oE25");
-    meEBe9oe25_ = dbe_->book1D(histo, histo, 100, 0.4, 1.1);
+  histo = "EB E4oE9";
+  meEBe4oe9_ = ib.book1D(histo, histo, 100, 0.4, 1.1);
 
-    sprintf(histo, "EB E16oE25");
-    meEBe16oe25_ = dbe_->book1D(histo, histo, 100, 0.4, 1.1);
-  }
-}
+  histo = "EBE9oE16";
```

Comment:
This line seems to have removed the space between EB E9oE16 giving the only difference in the Bin by bin comparison
We understand it is not intended.
Thank you @makortel

Reply:
Thanks for catching!